docs/APIReference-Renderer.md

```diff
@@ -0,0 +1,187 @@
+---
+id: api-reference-relay-renderer
+title: Relay.Renderer
+layout: docs
+category: API Reference
+permalink: docs/api-reference-relay-renderer.html
+next: api-reference-relay-root-container
+---
+
+**Relay.Renderer** is a replacement for `Relay.RootContainer` that composes a `Relay.ReadyStateRenderer` and performs data fetching for a given `queryConfig`. 
+
+## Overview
+
+*Props*
+
+<ul class="apiIndex">
+  <li>
+    <a href="#container">
+      <pre>Container</pre>
+      Relay container that defines fragments and the view to render.
+    </a>
+  </li>
+  <li>
+    <a href="#forcefetch">
+      <pre>forceFetch</pre>
+      Whether to send a server request regardless of data available on the client.
+    </a>
+  </li>
+  <li>
+    <a href="#onforcefetch">
+      <pre>onForceFetch</pre>
+      Called whenever forceFetch is being called
+    </a>
+  </li>
+  <li>
+    <a href="#onprimecache">
+      <pre>onPrimeCache</pre>
+      Called whenever primeCache is being called
+    </a>
+  </li>
+  <li>
+    <a href="#queryconfig">
+      <pre>queryConfig</pre>
+       QueryConfig or Route that defines the query roots.
+    </a>
+  </li>
+  <li>
+    <a href="#environment">
+      <pre>environment</pre>
+      The API for Relay core that provides methods for fetching/mutating data, an in-memory cache, etc.
+    </a>
+  </li>
+    <li>
+    <a href="#render">
+      <pre>render</pre>
+      Called to render when data requirements are being fulfilled.
+    </a>
+  </li>
+  <li>
+    <a href="#onreadystatechange">
+      <pre>onReadyStateChange</pre>
+    </a>
+  </li>
+</ul>
+
+## Props
+
+### Container
+
+```
+Container: RelayContainer
+```
+
+Must be a valid `RelayContainer`. Relay will attempt to fulfill its data requirements before rendering it.
+
+
+### forceFetch
+
+```
+forceFetch: boolean
+```
+
+If supplied and set to true, a request for data will always be made to the server regardless of whether data on the client is available to immediately fulfill the data requirements.
+
+
+### onForceFetch
+
+```
+onForceFetch(
+  querySet: RelayQuerySet,
+  callback: (readyState: ReadyState) => void
+): Abortable
+```
+
+This callback prop is called whenever a forceFetch occurs.
+
+
+### onPrimeCache
+
+```
+onPrimeCache(
+  querySet: RelayQuerySet,
+  callback: (readyState: ReadyState) => void
+): Abortable
+```
+
+This callback prop is called whenever a primeCache call occurs.
+
+### QueryConfig
+
+```
+queryConfig: RelayRoute
+```
+
+Either an instance of `Relay.Route` or an object with the `name`, `queries`, and optionally the `params` properties.
+
+
+### Environment
+
+```
+environment: RelayEnvironment
+```
+
+Either `Relay.Store`, an instance of `Relay.Evironment` or an object with the `forceFetch`, `getFragmentResolver`, `getStoreData`, `primeCache`, and `read` function defined.
```

Comment:
s/Evironment/Environment/

Rather than explicitly listing the methods (that may change over time), perhaps just say:

> An object that conforms to the `Relay.Environment` interface, such as `Relay.Store`.